includes/checkout/template.php

```diff
@@ -176,7 +176,7 @@ function edd_user_info_fields() {
 	$customer = array_map( 'sanitize_text_field', $customer );
 	?>
 	<fieldset id="edd_checkout_user_info">
-		<legend><?php echo apply_filters( 'edd_checkout_personal_info_text', esc_html__( 'Personal info', 'easy-digital-downloads' ) ); ?></legend>
+		<legend><?php echo apply_filters( 'edd_checkout_personal_info_text', esc_html_e( 'Personal info', 'easy-digital-downloads' ) ); ?></legend>
```

Comment:
Agree with @robincornett that:
* `esc_html__()` was the correct function
* `esc_html_e()` results in 2 `echo`'s

If an addon is filtering `edd_checkout_personal_info_text`, its return value will need to be translated using its own dedicated `textdomain` as registered. If it's expecting to use `easy-digital-downloads` for its `textdomain` it will need to perform additional work to do so.